src/main/java/seedu/address/logic/commands/ExportVolunteerCsvCommand.java

```diff
@@ -0,0 +1,114 @@
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.volunteer.Volunteer;
+
+/**
+ * Exports a person's volunteer information from SocialCare
+ */
+public class ExportVolunteerCsvCommand extends Command {
+    public static final String COMMAND_WORD = "exportvolunteercsv";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Exports a CSV file of the volunteer "
+            + "the specified index in the displayed volunteer list.\n"
+            + "You can specify more than one volunteer to add to the CSV "
+            + "by adding a whitespace after each index number\n"
+            + "Parameters: INDEX1 (must be a positive integer) INDEX2 INDEX3 ...\n"
+            + "Example 1: " + COMMAND_WORD + " 1 2 3 4 5 6 7";
+
+    private static final String MESSAGE_EXPORT_VOLUNTEER_SUCCESS = "Volunteer(s) exported as CSV "
+            + "to your Desktop.";
+    private static final String MESSAGE_EXPORT_VOLUNTEER_FAILED = "Volunteer(s) export failed, please try again.";
+
+    private final ArrayList<Index> index;
+
+    /**
+     * @param index of volunteer or event in the filtered list
+     */
+    public ExportVolunteerCsvCommand(ArrayList<Index> index) {
+        requireNonNull(index);
+        this.index = index;
+    }
+
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) throws CommandException {
+        requireNonNull(model);
+        List<Volunteer> list = model.getFilteredVolunteerList();
+
+        //Validate the given input within available index
+        for (Index i : index) {
+            // Handle case where the index input exceeds or equals the size of the last displayed list
+            if (i.getZeroBased() >= list.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_VOLUNTEER_DISPLAYED_INDEX);
+            }
+        }
+
+        try {
+            createVolunteerCsv(model.getFilteredVolunteerList(), index);
+        } catch (Exception e) {
+            throw new CommandException(MESSAGE_EXPORT_VOLUNTEER_FAILED);
+        }
+
+        return new CommandResult(MESSAGE_EXPORT_VOLUNTEER_SUCCESS);
+
+    }
+
+    /**
+     * Helper method to create and write the csv file given the volunteer
+     * @param list to contain the list of volunteers from model
+     * @param index to hold the list of indexes to export
+     */
+    private void createVolunteerCsv(ObservableList<Volunteer> list, ArrayList<Index> index) throws Exception {
+        // Setting up file path
+        File output = new File(System.getProperty("user.home") + "/Desktop/"
+                + Integer.toString(index.size()) + "volunteers.csv");
+
+        // Setting up writer & stringbuilder for appending
+        PrintWriter pw = new PrintWriter(output);
+        StringBuilder sb = new StringBuilder();
+        String csvSplit = ",";
```

Comment:
Might want to test this with fields that could contain commas, like Address

Reply:
Assumption on the csv entries dont have characters which messes with the tuple splits, like comma (,)and bracer ("").

Reply:
You should not work under this assumption. Users should be allowed to enter commas for addresses. Try to find a way to clean the input. Maybe this link would be helpful.

https://stackoverflow.com/questions/4617935/is-there-a-way-to-include-commas-in-csv-columns-without-breaking-the-formatting